core/templates/dev/head/expressions/expressionSyntaxTree.js

```diff
@@ -0,0 +1,420 @@
+    '>=': {
+      type: function(args, envs) {
+            verifyNumArgs(args, 2);
+            _verifyArgsHaveType(args, 'Real');
+            return 'UnicodeString';
+          },
+      eval: function(args, envs) {
+            verifyNumArgs(args, 2);
+            var numericArgs = _coerceAllArgsToNumber(args);
+            return numericArgs[0] >= numericArgs[1];
+          }
+    },
+    '<': {
+      type: function(args, envs) {
+           verifyNumArgs(args, 2);
+           _verifyArgsHaveType(args, 'Real');
+           return 'UnicodeString';
+         },
+      eval: function(args, envs) {
+           verifyNumArgs(args, 2);
+           var numericArgs = _coerceAllArgsToNumber(args);
+           return numericArgs[0] < numericArgs[1];
+         }
+    },
+    '>': {
+      type: function(args, envs) {
+           verifyNumArgs(args, 2);
+           _verifyArgsHaveType(args, 'Real');
+           return 'UnicodeString';
+         },
+      eval: function(args, envs) {
+           verifyNumArgs(args, 2);
+           var numericArgs = _coerceAllArgsToNumber(args);
+           return numericArgs[0] > numericArgs[1];
+         }
+    },
+    '!': {
+      type: function(args, envs) {
+           verifyNumArgs(args, 1);
+           _verifyArgsHaveType(args, 'UnicodeString');
+           return 'UnicodeString';
+         },
+      eval: function(args, envs) {
+           verifyNumArgs(args, 1);
+           return !args[0];
+         }
+    },
+    '==': {
+      type: function(args, envs) {
+            verifyNumArgs(args, 2);
+            return 'UnicodeString';
+          },
+      eval: function(args, envs) {
+            verifyNumArgs(args, 2);
+            return args[0] == args[1];
+          }
+    },
+    '!=': {
+      type: function(args, envs) {
+            verifyNumArgs(args, 2);
+            return 'UnicodeString';
+          },
+      eval: function(args, envs) {
+            verifyNumArgs(args, 2);
+            return args[0] != args[1];
+          }
+    },
+    '&&': {
+      type: function(args, envs) {
+            verifyNumArgs(args, 2);
+            _verifyArgsHaveType(args, 'UnicodeString');
+            return 'UnicodeString';
+          },
+      eval: function(args, envs) {
+            // TODO(kashida): Make this short-circuit.
+            verifyNumArgs(args, 2);
+            return Boolean(args[0] && args[1]);
+          }
+    },
+    '||': {
+      type: function(args, envs) {
+            verifyNumArgs(args, 2);
+            _verifyArgsHaveType(args, 'UnicodeString');
+            return 'UnicodeString';
+          },
+      eval: function(args, envs) {
+            // TODO(kashida): Make this short-circuit.
+            verifyNumArgs(args, 2);
+            return Boolean(args[0] || args[1]);
+          }
+    },
+    if: {
+      type: function(args, envs) {
+            verifyNumArgs(args, 3);
+            _verifyArgsHaveType([args[0]], 'UnicodeString');
+            _verifyArgsHaveType([args[2]], args[1]);
```

Comment:
Maybe better to add a new method: _verifyArgsHaveSameType

Also, would be good to add tests for this case, it's a tricky one